<?xml version="1.0"?>

<launch>

    <!--Controller Parameters-->
    <arg name="controller_kp" default="0.05"/>
    <arg name="controller_ki"  default="0.005"/>

    <!--Simulation Parameters-->
    <arg name="sample_time" default="0.02"/>
    <arg name="node_rate"  default="150"/>
    
    <!--Controller Limitations (Saturation)-->
    <arg name="max_output" default="1.0"/>
    <arg name="min_output" default="-1.0"/>

    <!--Initialise kinematic_model -->
    <node pkg="controller_simple" type="pid_controller.py" name="PID_controller" output="screen">
        <param name="controller_node_rate" value="$(arg node_rate)"/>
        <param name="controller_sample_time" value="$(arg sample_time)"/>
        <param name="controller_kp" value="$(arg controller_kp)"/>
        <param name="controller_ki" value="$(arg controller_ki)"/>
        <param name="controller_limit_min" value="$(arg min_output)"/>
        <param name="controller_limit_max" value="$(arg max_output)"/>        
    </node>

</launch>